pkg/plugin/bundle.go

```diff
@@ -28,11 +28,12 @@ type bundle struct {
 	plugins []Plugin
 
 	supportedProjectVersions []config.Version
+	deprecateWarning         string
 }
 
 // NewBundle creates a new Bundle with the provided name and version, and that wraps the provided plugins.
 // The list of supported project versions is computed from the provided plugins.
-func NewBundle(name string, version Version, plugins ...Plugin) (Bundle, error) {
+func NewBundle(name string, version Version, deprecateWarning string, plugins ...Plugin) (Bundle, error) {
```

Comment:
I wonder if converting this to use the options pattern would be better than adding another function parameter? I think that would prevent this function signature getting really long with future changes.

I think this would look something like:
```go
func NewBundle(opts ...BundleOptions) (Bundle, error)
```
and new bundles could be created like:
```go
bundle, err := NewBundle(
  WithName("name"),
  WithVersion(plugin.Version{Number: 4}),
  WithPlugins(plugins ...Plugin),
  WithDeprecationMessage("Deprecated!"),
)
```

Thoughts?

Reply:
I liked the idea a lot. 
What about we do that in a follow-up?

Reply:
@everettraven here we just fixing the method (which breaks the API)
I totally agree with the approach suggested by you and for we are able to do that I tracked the need in: https://github.com/kubernetes-sigs/kubebuilder/issues/3307